src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -48,8 +52,19 @@
      * <p>
      * Inserts an ItemStack into the given slot and return the remainder.
      * The ItemStack <em>should not</em> be modified in this function!
+     * Note: This behaviour is subtly different from {@link IFluidHandler#fill(FluidStack, IFluidHandler.FluidAction)}.
+     * </p>
+     * <p>
+     * insertItem must have the four following properties.
+     * First, the simulation mode must not mutate the inventory.
+     * Second, the simulation mode must be accurate, i.e insertItem(slot, stack, true)
```

Comment:
This is unfortunately not true, and mods need to deal with that case. Here is an example of an item handler that doesn't follow this guarantee: https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c0a9dd43df29f6edce051835bb9d662600497d80/src/main/java/appeng/parts/p2p/ItemP2PTunnelPart.java#L69.

(explanation: if an input P2P tunnel is connected to multiple output P2P tunnels on the same inventory, then simulation might allow inserting more than execution).